eventing/samples/writing-a-source/02-define-source.md

```diff
@@ -0,0 +1,202 @@
+# Define The Source Resource
+
+We'll use Kubebuilder to generate a new Kubernetes Custom Resource for our
+source. Check out the [Kubebuilder book](https://book.kubebuilder.io/) to learn
+more about Kubebuilder.
+
+## Create the Source CRD Skeleton
+
+_When in doubt, the
+[Kubebuilder Quick Start docs](https://book.kubebuilder.io/quick_start.html) are
+likely more current than these instructions._
+
+Use the `kubebuilder create api` command to create the API definition for a new
+CRD and a controller to reconcile it.
+
+You'll need to choose the following:
+
+*   **A group name.** This is the resource group that will contain the resource.
+    It's prepended to the domain name chosen earlier to produce the
+    fully-qualified resource name. The reference project uses `sources`.
+*   **A version name.** This is the initial version string for the CRD. It's
+    usually `v1alpha1` for new resources. The reference project uses `v1alpha1`.
+*   **A kind name.** This is the unqualified type name of the resource. The
+    reference project uses `SampleSource`.
+
+    The fully-qualified name of the reference resource is
+    `samplesources.sources.knative.dev`, and its `apiVersion` is
+    `sources.knative.dev/v1alpha1`.
+
+```sh
+kubebuilder create api --group sources --version v1alpha1 --kind SampleSource
+```
+
+This command will ask if you want to create the resource and controller. Answer
+yes to both to generate scaffolding to edit in future steps.
+
+The result of this command in the reference project can be viewed at
+https://github.com/grantr/sample-source/pull/2.
+
+## Add Fields to Source Type Definition
+
+_TODO explain why there are Sink and SinkURI fields_
```

Comment:
Do you want to make these markdown comments (`<!--` and `-->`), rather than visible in the rendered page?